helm/happa/values.yaml

```diff
@@ -109,3 +109,123 @@ containerSecurityContext:
     drop:
       - ALL
   runAsNonRoot: true
+
+# Gateway API HTTPRoute for happa frontend
+route:
+  # -- Enable Gateway API HTTPRoute for happa
+  enabled: false
+  # -- Set the route kind
+  # Valid options are GRPCRoute, HTTPRoute, TCPRoute, TLSRoute, UDPRoute
+  kind: HTTPRoute
+  # -- Override the route name (defaults to .Values.name)
+  name: ""
+  # -- Annotations to add to the route
+  annotations: {}
+  # -- Labels to add to the route
+  labels: {}
+  # -- Hostnames for the route (supports templating)
+  # Example: ["{{ .Values.happa.host }}"]
+  hostnames: []
+  # -- Parent gateway references
+  parentRefs: []
+  # - name: giantswarm-default
+  #   namespace: envoy-gateway-system
+  # -- Route matching rules
+  matches:
+    - path:
+        type: PathPrefix
+        value: /
+  # -- Filters define the filters that are applied to requests that match this rule
+  filters: []
+  # -- Additional custom rules that can be added to the route
+  additionalRules: []
+  # -- SecurityPolicy configuration (Envoy Gateway)
+  securityPolicy:
+    enabled: false
+    annotations: {}
+    labels: {}
+    # basicAuth:
+    #   users:
+    #     name: <secret_name>
+    # cors:
+    #   allowOrigins:
+    #     - "*"
+    #   allowMethods:
+    #     - GET
+    #     - POST
+    # jwt:
+    #   providers:
+    #     - name: example
+    #       issuer: https://example.com
+    # oidc:
+    #   provider:
+    #     issuer: https://example.com
+    # extAuth:
+    #   http:
+    #     service:
+    #       name: auth-service
+    #       port: 80
+    # authorization:
+    #   defaultAction: Deny
+    #   rules: []
+
+# Gateway API HTTPRoute for happaapi (kubernetes API proxy)
```

Comment:
since this is meant to proxy the kube api server (which has already certs), I think you'll need to add a BackendTLSPolicy

More info here https://gateway.envoyproxy.io/docs/tasks/security/backend-tls/